### PR TITLE
Fix too many args in Tvu::new

### LINKED ISSUE #836: Tvu::new() takes too many arguments
networking parameters could be collapsed, voting keypair is likely already carried in other parameters (crdt perhaps?)  clean up the signature of this function and others that take a similar pile of args

### PR DESCRIPTION
**Pull request by MarkJr94**
#### Problem
`Tvu::new`  takes too many arguments and is part of the public crate API

#### Summary of Changes
Now pass in sockets through the `solana::tvu::Sockets` struct

Move `ClusterInfo::keypair` to `pub(crate)` privacy  in order to remove redundant
signing keypair parameter

Fixes #836

### CODE REVIEW

File: src/fullnode.rs [hunk 1]
```diff
@@ -269,26 +269,33 @@ impl Fullnode {
...
+                fetch: node
+                    .sockets
+                    .tvu
+                    .iter()
+                    .map(|s| s.try_clone().expect("Failed to clone TVU Sockets"))
+                    .collect(),
+            };
+
+            let tvu = Tvu::new(
+                // keypair.clone(),
```
Comment (sakridge) [inline src/fullnode.rs]: remove commented code please

File: src/cluster_info.rs [hunk 2]
```diff
@@ -62,7 +62,7 @@ pub struct ClusterInfo {
     /// The network
     pub gossip: CrdsGossip,
     /// set the keypair that will be used to sign crds values generated. It is unset only in tests.
-    keypair: Arc<Keypair>,
+    pub(crate) keypair: Arc<Keypair>,
```
Comment (rob-solana) [inline src/cluster_info.rs]: I'm not familiar with this syntax

Comment (MarkJr94) [reply src/cluster_info.rs]: Removed in commit d309cfd

File: src/fullnode.rs (hunk 1 above)
Comment (MarkJr94) [reply src/fullnode.rs]: Removed in commit d309cfd

**Comment by rob-solana** [comment]
thanks MarkJr94 for this contribution!

**Pull request merged**